test/application/widgets/theme/highlightable_text_test.dart

```diff
@@ -0,0 +1,63 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:memo/application/widgets/theme/highlightable_text.dart';
+
+import '../../../utils/widget_pump.dart';
+
+void main() {
+  const fakeText = 'A fake text';
+  const fakeTextStyle = TextStyle(color: Colors.white);
+  const fakeHighlight = 'fake';
+  const fakeHighlightStyle = TextStyle(color: Colors.red);
+
+  testWidgets('should not style highlight prefix', (tester) async {
+    const highlightableText = HighlightableText(
+      text: fakeText,
+      textStyle: fakeTextStyle,
+      highlighted: fakeHighlight,
+      highlightedStyle: fakeHighlightStyle,
+    );
+    const expectedPrefixSpan = TextSpan(text: 'A ', style: fakeTextStyle);
+
+    await pumpProviderScoped(tester, highlightableText);
+
+    final richText = tester.widget(find.byType(RichText)) as RichText;
+    final textSpan = richText.text as TextSpan;
+    final innerTextSpan = textSpan.children!.first as TextSpan;
+    expect(innerTextSpan.children!.first, expectedPrefixSpan);
+  });
+
+  testWidgets('should style highlight text', (tester) async {
+    const highlightableText = HighlightableText(
+      text: fakeText,
+      textStyle: fakeTextStyle,
+      highlighted: fakeHighlight,
+      highlightedStyle: fakeHighlightStyle,
+    );
+    const expectedPrefixSpan = TextSpan(text: 'fake', style: fakeHighlightStyle);
+
+    await pumpProviderScoped(tester, highlightableText);
+
+    final richText = tester.widget(find.byType(RichText)) as RichText;
+    final textSpan = richText.text as TextSpan;
+    final innerTextSpan = textSpan.children!.first as TextSpan;
+    expect(innerTextSpan.children![1], expectedPrefixSpan);
+  });
+
+  testWidgets('should not style highlight suffix', (tester) async {
+    const highlightableText = HighlightableText(
+      text: fakeText,
+      textStyle: fakeTextStyle,
+      highlighted: fakeHighlight,
+      highlightedStyle: fakeHighlightStyle,
+    );
+    const expectedPrefixSpan = TextSpan(text: ' text', style: fakeTextStyle);
+
+    await pumpProviderScoped(tester, highlightableText);
+
+    final richText = tester.widget(find.byType(RichText)) as RichText;
+    final textSpan = richText.text as TextSpan;
+    final innerTextSpan = textSpan.children!.first as TextSpan;
+    expect(innerTextSpan.children!.last, expectedPrefixSpan);
+  });
```

Comment:
```suggestion
    const highlightableText = HighlightableText(
      text: fakeText,
      textStyle: fakeTextStyle,
      highlighted: fakeHighlight,
      highlightedStyle: fakeHighlightStyle,
    );
    
     await pumpProviderScoped(tester, highlightableText);
     
    final richText = tester.widget(find.byType(RichText)) as RichText;
    final textSpan = (richText.text as TextSpan).children!.first as TextSpan;

  testWidgets('should use the default textStyle on anything other than the highlightable text', (tester) async {
    const expectedPrefixSpan = TextSpan(text: 'A ', style: fakeTextStyle);
    expect(textSpan.children![0], expectedPrefixSpan);
    
    const expectedSuffixSpan = TextSpan(text: ' text', style: fakeTextStyle);
    expect(textSpan.children![2], expectedSuffixSpan);
  });
  
  testWidgets('should use the highlightedStyle on highlighted text', (tester) async {
    const expectedHighlightedSpan = TextSpan(text: 'fake', style: fakeHighlightStyle);
    expect(textSpan.children![1], expectedHighlightedSpan);
  });
```

Reply:
ofc you won't have tester outside that scope, but you can work around that

Reply:
Added a function to to that. Let me know what do you think.